src/Starcounter.Database.Extensions/PostCommitTransactor.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Starcounter.Database.ChangeTracking;
+
+namespace Starcounter.Database.Extensions
+{
+    public class PostCommitTransactor : TransactorBase
+    {
+        readonly AsyncLocal<List<KeyValuePair<Type, Change>>> _lastChanges = new AsyncLocal<List<KeyValuePair<Type, Change>>>();
+        readonly PostCommitOptions _hookOptions;
+
+        public PostCommitTransactor(ITransactor transactor, IOptions<PostCommitOptions> postCommitHookOptions)
+            : base(transactor)
+            => _hookOptions = postCommitHookOptions.Value;
+
+        protected override void LeaveContext(IDatabaseContext db, bool exceptionThrown)
+        {
+            List<KeyValuePair<Type, Change>> changes = new List<KeyValuePair<Type, Change>>();
+
+            foreach (var change in db.ChangeTracker.Changes.Where(c => c.Type != ChangeType.Delete))
```

Comment:
Doesn't pre-commit transactor run the same changes enumeration? Worth to extract in a helper then?

Reply:
Most likely.